SpokestackTests/TFLiteWakewordRecognizerTest.swift

```diff
@@ -0,0 +1,234 @@
+        config.detectModelPath = MockWakewordModels.detectPath
+        tflwr.configuration = config
+        tflwr.context = context
+        tflwr.delegate = delegate
+        
+        // start
+        context.isActive = false
+        tflwr.startStreaming(context: context)
+        tflwr.activate(frame: Frame.voice(frameWidth: 10, sampleRate: 8000))
+        XCTAssert(context.isSpeech)
+        tflwr.deactivate()
+        XCTAssertFalse(context.isSpeech)
+    }
+    
+    func testProcess() {
+        // setup
+        let tflwr = TFLiteWakewordRecognizer.sharedInstance
+        let context = SpeechContext()
+        let delegate = TFLiteWakewordRecognizerTestDelegate()
+        let config = SpeechConfiguration()
+        let successExpectation = XCTestExpectation(description: "process without failure.")
+        delegate.didActivateExpectation = successExpectation
+        config.encodeModelPath = MockWakewordModels.encodePath
+        config.filterModelPath = MockWakewordModels.filterPath
+        config.detectModelPath = MockWakewordModels.detectPath
+        config.vadMode = .HighlyPermissive
+        tflwr.configuration = config
+        tflwr.context = context
+        tflwr.delegate = delegate
+        
+        // process
+        context.isActive = false
+        context.isSpeech = false
+        // NB: the detect model will always output 1.0 no matter the input
+        tflwr.process(Frame.voice(frameWidth: 10, sampleRate: 16000))
+        wait(for: [successExpectation], timeout: 5)
+        XCTAssert(context.isActive)
+        XCTAssertFalse(delegate.didError)
+    }
+}
+
+fileprivate enum MockWakewordModels {
+    static let filterInfo = (name: "mock_filter", extension: "tflite")
+    static let encodeInfo = (name: "mock_encoder", extension: "tflite")
+    static let detectInfo = (name: "mock_detector", extension: "tflite")
+    static let input = [Int32](Array(repeating: 0, count: 128)).withUnsafeBufferPointer(Data.init)
+    static let filterInput = [Int32](Array(repeating: 0, count: 257)).withUnsafeBufferPointer(Data.init)
+    static let encodeInput = [Int32](Array(repeating: 0, count: 40)).withUnsafeBufferPointer(Data.init)
+    static let detectInput = [Int32](Array(repeating: 0, count: 12800)).withUnsafeBufferPointer(Data.init)
+    static let validIndex = 0
+    static let shape: TensorShape = [2]
+    static let output = [Int32](Array(repeating: 0, count: 40))
+    static let outputData = output.withUnsafeBufferPointer(Data.init)
+    static let filterOutput = [Int32](Array(repeating: 0, count: 40))
+    static let filterOutputData = filterOutput.withUnsafeBufferPointer(Data.init)
+    static let encodeOutput = [Int32](Array(repeating: 0, count: 128))
+    static let encodeOutputData = encodeOutput.withUnsafeBufferPointer(Data.init)
+    static let detectOutput = [Int32](arrayLiteral: 1065353216)
+    static let detectOutputData = detectOutput.withUnsafeBufferPointer(Data.init)
+    static var filterPath: String = {
```

Comment:
Given the multiple usages here and in the other TF test, this could probably go in `SharedTestMocks`. Not sure the `(filter|encode|detect)Info` need to be separate from these path creations, either.

Reply:
i think there's a better way to structure this (it's an artifact from the tensorflowlite-swift testing approach), but I'll leave it for a future refactor.